projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/upload/JobUploadFileFilter.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2021-2023 VMware, Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.vmware.taurus.service.upload;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.stream.Stream;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+/**
+ * This class filters out the job directory from forbidden files so that it can be uploaded without
+ * them. Forbidden files are specified in a comma separated list in apache tika format. Full list of
+ * supported files can be found - https://tika.apache.org ; This operation is intended to allow the
+ * upload of a data job by deleting any files specified in the:
+ * upload.validation.fileTypes.filterlist property.
+ */
+@Service
+@Slf4j
+public class JobUploadFileFilter {
+
+  private final String[] filterList;
+  private FileFormatDetector formatDetector;
+
+  public JobUploadFileFilter(
+      @Value("${upload.validation.fileTypes.filterlist:}") String[] filterList) {
+    this.filterList = filterList;
+    this.formatDetector = new FileFormatDetector();
+  }
+
+  /**
+   * Data job directory to be filtered from files present in the
+   * upload.validation.fileTypes.filterlist variable. All matching files are deleted from the
+   * directory and sub-directories before being uploaded to git version control.
+   *
+   * @param jobFolder
+   * @param jobName
+   */
+  public void filterDirectory(File jobFolder, String jobName) {
+
+    if (this.filterList.length == 0) {
+      return;
+    }
+
+    try (Stream<Path> stream = Files.walk(Paths.get(jobFolder.getPath()))) {
+      stream
+          .filter(Files::isRegularFile)
+          .forEach(
+              file -> {
+                validateFile(file.toAbsolutePath(), jobName);
+              });
+    } catch (IOException e) {
+      log.error("Exception while processing filter list: {}", e);
```

Comment:
We are actually trying to clean up error logging in our system at the moment. 
We are seeing the same error logged many times. 
Can you please not log it here. 

The new rule is that you should only log an error if you are going to recover from it. 
Sprint will log it at the exit point of the web request so there is no fear that the exception won't get logged. 

It might just be cleaner to let the exception bubble up instead of trying to handle it at all.